pkg/operator/cluster/ceph/mon/spec.go

```diff
@@ -55,47 +57,56 @@ func AdminSecretEnvVar() v1.EnvVar {
 	return v1.EnvVar{Name: "ROOK_ADMIN_SECRET", ValueFrom: &v1.EnvVarSource{SecretKeyRef: ref}}
 }
 
-func (c *Cluster) getLabels(name string) map[string]string {
+// PodNameEnvVar sets the Pod name as the name
+func PodNameEnvVar() v1.EnvVar {
+	return v1.EnvVar{Name: "ROOK_NAME", ValueFrom: &v1.EnvVarSource{
+		FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.name"},
+	}}
+}
+
+func (c *Cluster) getLabels() map[string]string {
 	return map[string]string{
 		k8sutil.AppAttr: appName,
-		"mon":           name,
 		monClusterAttr:  c.Namespace,
 	}
 }
 
-func (c *Cluster) makeReplicaSet(config *monConfig, hostname string) *extensions.ReplicaSet {
-	rs := &extensions.ReplicaSet{
+func (c *Cluster) makeStatefulSet(replicas int32) *v1beta1.StatefulSet {
+	pod := c.makeMonPod()
+	sts := &v1beta1.StatefulSet{
```

Comment:
use v1 api

Reply:
@rootfs Is the v1 api compatible to older K8s versions (1.6, 1.7, 1.8) without that API version?